packages/studio/src/components/TailwindPropInput.tsx

```diff
@@ -0,0 +1,189 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import generateTailwindSafelist from "../utils/generateTailwindSafelist";
+import { StudioTailwindTheme } from "@yext/studio-plugin";
+import { ReactComponent as EmbedIcon } from "../icons/embed.svg";
+import { ReactComponent as X } from "../icons/x.svg";
+import { useRootClose } from "@restart/ui";
+import classNames from "classnames";
+
+const customClassesPromise: Promise<string[] | undefined> = import(
+  "@pathToUserProjectRoot/tailwind.config"
+)
+  .then((module) => {
+    const theme: StudioTailwindTheme | undefined =
+      module.default?.theme?.extend;
+    return theme && generateTailwindSafelist(theme);
+  })
+  .catch((e) => {
+    console.error(e);
+    return undefined;
+  });
+
+interface Props {
+  onChange: (value: string) => void;
+  value: string;
+  disabled: boolean;
+}
+
+export default function TailwindPropInput({
```

Comment:
do we also want to support typing into the input so people can specify non-custom Tailwind classes?

Reply:
We don't support non-custom tailwind classes right now so not yet. Right now somebody may type a tailwind class that isn't included in the current tailwind bundle on the page